### ISSUE TITLE
useless-return and explicit "return None"

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#2300]
Similarly to what's already been said [on the mailinglist](https://mail.python.org/pipermail/code-quality/2018-July/001037.html), I don't think an explicit `return None` should count as useless return.

Consider this class:

```python
class PseudoLayout(QLayout):

    # [...]

    def count(self) -> int:
        return 0

    def itemAt(self, _pos) -> Optional[QWidget]:
        return None
````

Here, I do need to override `itemAt` (because PyQt or rather C++ tells me to), but my layout never actually has any items in it, so I always want to `return None`.

Sure, from Python's perspective, I might as well just put `pass` there and rely on the implicit `return None`, but IMHO this is a case of "explicit is better than implicit". The function is clearly intended to return something (and not just there for it's side effects), so IMHO, there *should* be an explicit `return` here.

**Comment by PCManticore** [comment]
Thanks for the report, should be fixed now!

**Comment by gallamine** [comment]
I am still receiving this on `pylint==2.1.1`.

**Comment by The-Compiler** [comment]
I don't. If you're sure you're running 2.1.1, you might want to open a new issue and include more information (like a minimal example).
